presto-main/src/test/java/io/prestosql/server/TestQueryResource.java

```diff
@@ -111,6 +144,49 @@ public void testGetQueryInfoExecutionFailure()
         assertEquals(info.getFailureInfo().getErrorCode(), DIVISION_BY_ZERO.toErrorCode());
     }
 
+    @Test
+    public void testCancel()
+    {
+        String queryId = startQuery("SELECT * FROM tpch.sf100.lineitem");
```

Comment:
You could introduce a testing procedure or function to sleep instead of wasting CPU.

Reply:
This technique is used by several tests (I copied it from somewhere else).  I agree they should all be update, but that seems like an orthogonal project.